test/test_puma_server.rb

```diff
@@ -427,20 +432,16 @@ def test_lowlevel_error_message
     assert_match(/Puma caught this error: Oh no an error.*\(NoMethodError\).*test\/test_puma_server.rb/m, data)
   end
 
-  class WithoutBacktraceError < StandardError
-    def backtrace; nil; end
-    def message; "no backtrace error"; end
-    def class; "WithoutBacktraceError"; end
-  end
-
   def test_lowlevel_error_message_without_backtrace
     @server = Puma::Server.new @app, @events, {log_writer: @log_writer, :force_shutdown_after => 2}
 
     server_run do
       raise WithoutBacktraceError.new
     end
 
-    data = send_http_and_read "GET / HTTP/1.1\r\n\r\n"
+    skt = send_http "GET / HTTP/1.1\r\n\r\n"
+    sleep 1.0 if TRUFFLE
+    data = skt.sysread 2_048
```

Comment:
Can you explain why this helps? Maybe @eregon is interested to hear about this workaround